src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -210,6 +221,66 @@ def run(
                 )
                 raise unexpected_error from e
 
+    def _copy_previous_run_result(
+        self,
+        resume_from_run_storage: AbstractBatchRunStorage,
+        resume_from_run_output_dir: Path,
+        batch_inputs: List,
+    ) -> List[LineResult]:
+        """Duplicate the previous debug_info from resume_from_run_storage and output from resume_from_run_output_dir
+        to the storage of new run,
+        return the list of previous line results for the usage of aggregation and summarization.
+        """
+        previous_run_results = []
+        previous_run_output = self._load_outputs(resume_from_run_output_dir) if resume_from_run_output_dir else []
+        previous_run_output_dict = {
+            each_line_output[LINE_NUMBER_KEY]: each_line_output for each_line_output in previous_run_output
+        }
+        for i in range(len(batch_inputs)):
+            previous_run_info = resume_from_run_storage.load_flow_run_info(i) if resume_from_run_storage else None
+            if previous_run_info and previous_run_info.status == Status.Completed:
+                previous_node_run_infos = (
+                    resume_from_run_storage.load_node_run_info_for_line(i) if resume_from_run_storage else []
+                )
+                previous_node_run_infos_dict = {node_run.node: node_run for node_run in previous_node_run_infos}
+                previous_node_run_outputs = {node_info.node: node_info.output for node_info in previous_node_run_infos}
+                aggregation_inputs = extract_aggregation_inputs(self._flow, previous_node_run_outputs)
+                self._storage.persist_flow_run(previous_run_info)
+                for node_run_info in previous_node_run_infos:
+                    self._storage.persist_node_run(node_run_info)
+                previous_line_result = self._construct_line_result(
```

Comment:
I know I wrote this function in the first place...
But the logic seems not need another function, please just move the logic here :)